common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -157,6 +159,75 @@ class @Problem
 
   # TODO add hooks for problem types here by inspecting response.html and doing
   # stuff if a div w a class is found
+          if response.progress_status != "done"
+            @maybeSetupTimer()
+
+  maybeSetupTimer: =>
+    if @el.find(".problem-timer").length
+      @$timer = @el.find('.problem-timer')
+      @$display = @el.find('.minutes-left')
+      @seconds_left = parseInt(@$timer.data('secondsLeft'), 10)
+      @seconds_before_warning = 60 * parseInt(@$timer.data('minutesBeforeWarning'), 10)
+      @submitted_before_time_expired = false
+
+      # The timer is initially hidden by CSS 
+      @$timer.show()
+
+      # Clear old timers, eg. if we press the submit button
+      if @timer_id
+        clearInterval(@timer_id)
+
+      # Sync every second
+      @timer_id = setInterval(@syncTimer, 1000)
+
+      # Initialize and show timer
+      @syncTimer()
+
+  syncTimer: =>
+    @seconds_left -= 1
```

Comment:
by subtracting 1 second every timer tick, are you depending on the precision of the `setInterval` to be exactly 1 second if you set it to 1000ms?  I don't think that's reliable due to JS being single threaded.  Would a better approach be to record the local browser timestamp when `maybeSetupTimer` is first called, and then every time syncTimer fires, get a request fresh timestamp from the browser, and compare it to the original, and use that elapsed time to compute seconds left.  That should make the implementation not dependent on the precision of `setInterval`.